### PR TITLE
add bmm shape inference

### PR DESCRIPTION
**Pull request by Yancey1989**

### CODE REVIEW

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 1]
```diff
@@ -2303,8 +2303,20 @@ class ShapePropagator : public PropertyPropBase {
...
+        std::vector<ShapeSymbol> new_sizes =
+            t1->symbolic_sizes().sizes().value();
+        new_sizes[2] = t2->symbolic_sizes().sizes().value()[2];
+        node->output()->setType(t1->withSymbolicShapes(new_sizes));
+      }
+      return true;
+    } else if (
+        node->matches("aten::pow(Tensor self, Scalar exponent) -> Tensor") ||
+        node->matches(
+            "aten::gelu(Tensor self, *, str approximate='none') -> Tensor")) {
```
Comment (tanyokwok) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:2319]: Nit: you can register `pow` and `gelu` to `first_input_type_formula` at Line 873 directly

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 2]
```diff
@@ -2303,8 +2303,20 @@ class ShapePropagator : public PropertyPropBase {
         }
       }
       return success;
-    } else if (node->matches(
-                   "aten::pow(Tensor self, Scalar exponent) -> Tensor")) {
+    } else if (node->matches("aten::bmm(Tensor self, Tensor mat2) -> Tensor")) {
+      auto t1 = tensor_types.at(0);
+      auto t2 = tensor_types.at(1);
+      if (t1 && t2) {
```
Comment (tanyokwok) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:2309]: Nit: emit a warning if the rank of t1 and t2 are not equals to 3?

**Review by tanyokwok — APPROVED** [review]

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp (hunk 1 above)
Comment (Yancey1989) [reply pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp]: done.

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp (hunk 2 above)
Comment (Yancey1989) [reply pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp]: pass the shape inference for op with `rank != 3`

**Review by tanyokwok — APPROVED** [review]
LGTM

**Pull request merged**